### PR TITLE
Refactor gateway test assertions into `test_auth` function

### PR DESCRIPTION
**Pull request by elsirion**

### CODE REVIEW

**Review by okjodom — APPROVED** [review]
Concise! Looking good :)

File: gateway/tests/tests/tests.rs [hunk 1]
```diff
@@ -129,20 +102,21 @@ async fn test_gateway_authentication() -> Result<()> {
...
-    assert_ne!(
-        client
-            .withdraw(gw_password.clone(), payload,)
-            .await?
-            .status(),
+        // use random password here
+        func("foobar123456789".to_string()).await?.status(),
         401
     );
+    assert_ne!(func(gw_password.to_string()).await?.status(), 401);
```
Comment (jkitman) [inline gateway/tests/tests/tests.rs:119]: Maybe this should be eq `200` just to be more explicit.

**Review by jkitman — APPROVED** [review]
Testing auth on every endpoint is maybe a bit overkill, but easier to read with the refactor.

File: gateway/tests/tests/tests.rs (hunk 1 above)
Comment (elsirion) [reply gateway/tests/tests/tests.rs:119]: That actually confused me too, but then it felt like a unrelated change. Let me add a commit to fix it.

Comment (okjodom) [reply gateway/tests/tests/tests.rs:119]: 200 would be testing auth, and all the other logic of the call.

Comment (elsirion) [reply gateway/tests/tests/tests.rs:119]: `get_balance` fails with a 500 error, leaving it to @okjodom to fix if possible. Maybe this test is actually just testing that authentication does or doesn't fail and is intended to error internally.

File: gateway/tests/tests/tests.rs [hunk 2]
```diff
@@ -129,20 +102,21 @@ async fn test_gateway_authentication() -> Result<()> {
         amount: bitcoin::Amount::from_sat(100),
         address: bitcoin.get_new_address(),
     };
+    test_auth(&gw_password, |pw| client_ref.withdraw(pw, payload.clone())).await?;
+
+    task_group.shutdown_join_all().await
+}
+
+async fn test_auth<Fut>(gw_password: &str, func: impl Fn(String) -> Fut) -> Result<()>
```
Comment (dpc) [inline gateway/tests/tests/tests.rs:110]: Docstrings please! :D

"Test that a given endpoint/functionality fails with the wrong password but works with the correct one?" :)

Comment (elsirion) [reply gateway/tests/tests/tests.rs:110]: :sweat_smile:

**Pull request merged**